File: veileder-flate/src/mock/entities/tiltakstype.ts
import { Entity } from '@mswjs/data/lib/glossary';
import { Tiltakskode, Tiltakstype } from 'mulighetsrommet-api-client';
import { DatabaseDictionary } from '../database';

export type TiltakstypeEntity = Entity<DatabaseDictionary, 'tiltakstype'>;

export function toTiltakstype(entity: TiltakstypeEntity): Tiltakstype {
  return {
    id: entity.id,
    innsatsgruppe: entity.innsatsgruppe?.id ?? null,
    navn: entity.navn,
    tiltakskode: Tiltakskode[entity.tiltakskode as keyof typeof Tiltakskode],
  };
}

File: mulighetsrommet-api/server/src/main/kotlin/no/nav/mulighetsrommet/api/tiltakstype/model/Tiltakstype.kt
package no.nav.mulighetsrommet.api.tiltakstype.model

import no.nav.mulighetsrommet.model.Innsatsgruppe
import no.nav.mulighetsrommet.model.Tiltakskode
import java.util.UUID

data class Tiltakstype(
    val id: UUID,
    val navn: String,
    val innsatsgrupper: Set<Innsatsgruppe>,
    val tiltakskode: Tiltakskode,
    val arenakode: String?,
    val sanityId: UUID?,
)
